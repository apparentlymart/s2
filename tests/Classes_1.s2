#-*-c++-*-

layerinfo "type" = "core";
layerinfo "majorversion" = 0;

property string blah;
set "blah" = "foo";

class Dog
{
  var string name;
  function out;
}

class KillerDog extends Dog
{
  var int weight;
}

class Page
{
  var KillerDog d;
  function foo (int c);
}


function Dog::out {
  println "I'm a dog named \"$.name\".";
}

function KillerDog::out {
  println "I'm a killer dog named \"$.name\" with weight $.weight";
}

function main ()
{
  var KillerDog kd = new KillerDog;
  $kd.name = "Killer Dog";
  $kd.weight = 9999;
  $kd->out();
}

