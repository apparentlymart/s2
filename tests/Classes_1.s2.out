I'm a killer dog named "Killer Dog" with weight 9999
